dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/trackedentity/store/DefaultAclStoreTest.java

```diff
@@ -0,0 +1,181 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.dxf2.events.trackedentity.store;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasSize;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.hisp.dhis.IntegrationTestBase;
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.UserGroup;
+import org.hisp.dhis.user.UserService;
+import org.hisp.dhis.user.sharing.UserAccess;
+import org.hisp.dhis.user.sharing.UserGroupAccess;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class DefaultAclStoreTest
+    extends IntegrationTestBase
+{
+
+    @Autowired
+    private UserService _userService;
+
+    @Autowired
+    private DefaultAclStore aclStore;
+
+    @Autowired
+    IdentifiableObjectManager manager;
+
+    private User owner;
+
+    private User user;
+
+    @Before
+    public void setUp()
+    {
+        // DhisConvenienceTests needs it to be injected/set for createUser
+        userService = _userService;
+        user = createUser( "userWithoutUserGroup" );
+        owner = createUser( "owner" );
+    }
+
+    @Test
+    public void getAccessibleProgramsReturnsNoneIfNoneIsPublicAndUserHasNoAccess()
+    {
+
+        // a private program
+        Program programA = createProgram( 'A' );
+        programA.setPublicAccess( "--------" );
+        programA.getSharing().setOwner( owner );
+        manager.save( programA, false );
+        // a private program readable by a user group of which the user is NOT
+        // part of
+        Program programB = createProgram( 'B' );
+        programB.setPublicAccess( "--------" );
+        programB.getSharing().setOwner( owner );
+        Set<User> users = new HashSet<>();
+        users.add( owner );
+        UserGroup g = createUserGroup( 'B', users );
+        UserGroupAccess a = new UserGroupAccess();
+        a.setUserGroup( g );
+        a.setAccess( "--r-----" );
+        programB.getSharing().addUserGroupAccess( a );
+        manager.save( programB, false );
+
+        List<Long> programIds = aclStore.getAccessiblePrograms( user.getUid(), Collections.emptyList() );
+
+        assertThat( programIds, hasSize( 0 ) );
```

Comment:
Most of the codebase does not use `assertThat` style.
I think we should stick to `assertEquals` and co where possible.

Reply:
Do you mean as long as I can assert using junit provided assertions I should use them and only resort to hamcrest if I need one of their provided matchers? Fine by me 😄 It's just never easy to stick to a convention without for example a linter  enforcing it :grin: If you search the codebase for `assertThat(` you'll find >1500 results, quite a lot.

Reply:
so here you prefer?

```java
assertNotNull(programIds);
assertEquals(programIds.size(), 1);
```

currently it will handle programIds being null

assertContainsOnly would also cause an NPE if the collection passed in is null. I can create a PR for handling that. Bit cumbersome having to assert that before as well, from my point of view.

Reply:
hamcrest also has contains, containsInAnyOrder http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#contains%28E...%29 handling null, creating nice assertion messages

Somehow I did not find it when importing. That's actually what I wanted to use 😅